### PR TITLE
【Hackathon + No.225】 + 提交作品

### PR DESCRIPTION
**Pull request by SOVLOOKUP**
## overview
- [x] [No.225](https://github.com/PaddlePaddle/Paddle/issues/50632#task225) https://github.com/jina-ai/jina-paddle-hackathon/issues/2 
image

## jina hub:
https://cloud.jina.ai/executor/2aalq6xu

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (numb3r3) [inline .vscode/settings.json]: ignore `.vscode` in the git repo

File: README.md [hunk 2]
```diff
@@ -1 +1,12 @@
 # 极纳 x 百度飞桨黑客马拉松
+
+## 实现
+
+1. [src/uie](./src/uie/) Jina 的 UIE Executor & Flow 实现、功能测试、可视化页面
+
+![](https://user-images.githubusercontent.com/53158137/221882155-05a23b18-2007-4321-a0bb-961eebe7439b.png)
+![](https://user-images.githubusercontent.com/53158137/221882528-14070e16-4829-4552-ad7e-88fe78ea1e23.png)
+
+2. [src/plato-mini](./src/plato-mini/) Jina 的 Plato-Mini Executor & Flow 实现、功能测试
```
Comment (numb3r3) [inline README.md]: It seems that the second item is not relevant to this ticket.

File: src/uie/.pdm.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[python]
```
Comment (numb3r3) [inline src/uie/.pdm.toml]: ignore the local pdm toml file.

File: src/uie/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+## Jina X UIE
+
+## GetStart
+
+please install pdm first: `pip install pdm`
```
Comment (numb3r3) [inline src/uie/README.md:5]: Please provide more information about `pdm`.

File: src/uie/pdm.lock [hunk 5]
```diff
@@ -0,0 +1,3140 @@
+# This file is @generated by PDM.
```
Comment (numb3r3) [inline src/uie/pdm.lock]: ignore this file as well

File: src/uie/src/uie_executor/config.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+jtype: UIEExecutor
+py_modules:
+  - executor.py
+metas:
+  name: UIEExecutor
+  description: Paddle UIE Executor
+  url: https://github.com/jina-ai/jina-paddle-hackathon
+  keywords: ['nlp']
```
Comment (numb3r3) [inline src/uie/src/uie_executor/config.yml]: refine the keywords by providing more relevant tags.

File: src/uie/src/uie_executor/executor.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    # ['时间', '赛手', '赛事名称', '情感倾向[正向，负向]']  实体抽取 分类任务需要[]来设置分类的label
+    str,
+    # {'歌曲名称': ['歌手', '所属专辑']} 实体属性抽取
+    Dict[str, List[str]]
+]
+
+
+class UIEExecutor(Executor):
+    def __init__(self,  **kwargs):
+        super().__init__(**kwargs)
```
Comment (numb3r3) [inline src/uie/src/uie_executor/executor.py:15]: The default `schema` can be passed as a parameter.

File: src/uie/src/uie_executor/executor.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        self.uie = Taskflow('information_extraction',
+                            schema=['时间', '选手', '赛事名称'])
+
+    @requests
+    async def call_uie(self, docs, parameters: Dict[str, List[Schema]], **kwargs):
+        if (parameters["schema"] != None):
+            self.uie.set_schema(parameters["schema"])
+
+        for d in docs:
+            d.text = str(self.uie(d.text))
```
Comment (numb3r3) [inline src/uie/src/uie_executor/executor.py:24]: Should we have a structured response? For example, to return the result in `.tags`:
```
result_doc.tags['时间'] = ...
result_doc.tags['选手'] = ...
...
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Comment by SOVLOOKUP** [comment]
Thanks for your advice!

**Review by numb3r3 — APPROVED** [review]
LGTM

**Pull request merged**